manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -303,28 +303,27 @@ func (r *FybrikApplicationReconciler) deleteExternalResources(applicationContext
 	return nil
 }
 
-// setReadModulesEndpoints populates the ReadEndpointsMap map in the status of the fybrikapplication
-func setReadModulesEndpoints(application *api.FybrikApplication, flows []api.Flow) {
-	readEndpointMap := make(map[string]taxonomy.Connection)
+// setVirtualEndpoints populates the endpoints in the status of the fybrikapplication
+func setVirtualEndpoints(application *api.FybrikApplication, flows []api.Flow) {
+	endpointMap := make(map[string]taxonomy.Connection)
 	for _, flow := range flows {
-		if flow.FlowType == taxonomy.ReadFlow {
-			for _, subflow := range flow.SubFlows {
-				if subflow.FlowType == taxonomy.ReadFlow {
-					for _, sequentialSteps := range subflow.Steps {
-						// Check the last step in the sequential flow that is for read (this will expose the reading api)
-						lastStep := sequentialSteps[len(sequentialSteps)-1]
-						if lastStep.Parameters.API != nil {
-							readEndpointMap[flow.AssetID] = lastStep.Parameters.API.Connection
-						}
-					}
-				}
+		// sanity check
+		if len(flow.SubFlows) == 0 {
+			continue
+		}
+		subflow := flow.SubFlows[len(flow.SubFlows)-1]
+		for _, sequentialSteps := range subflow.Steps {
+			// Check the last step in the sequential flow (this will expose the api)
+			lastStep := sequentialSteps[len(sequentialSteps)-1]
+			if lastStep.Parameters.API != nil {
+				endpointMap[flow.AssetID] = lastStep.Parameters.API.Connection
 			}
 		}
 	}
 	// populate endpoints in application status
 	for _, asset := range application.Spec.Data {
 		state := application.Status.AssetStates[asset.DataSetID]
-		state.Endpoint = readEndpointMap[asset.DataSetID]
+		state.Endpoint = endpointMap[asset.DataSetID]
 		application.Status.AssetStates[asset.DataSetID] = state
```

Comment:
is it enough to do `application.Status.AssetStates[asset.DataSetID].Endpoint = endpointMap[asset.DataSetID]` instead of the three lines above?

Reply:
It's a Go restriction that prevents you from updating a field of a map.